integration-tests/test-helpers.ts

```diff
@@ -0,0 +1,19 @@
+import { TramWindow } from '../src/types';
+
+const { waitFor } = require('@testing-library/dom');
+const { startApp } = require('./test-app');
+
+/**
+ * decorated startApp function that ensures that the app's mutation observers
+ * have kicked in before starting to interact with the app
+ */
```

Comment:
often times in the unit tests the mutation observers would try to handle multiple interactions at the same time. This did not occur when using the test-app locally, but was a side-effect of jest/testing-library not waiting for the mutation observer to catch up as it was interacting with the app (and, to be fair, rightfully so). 

This decorated `startApp` function starts the app, and then waits for one of our stores (which is updated by the mutation observer) to be populated. This allowed the tests to pass as expected.